src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -71,12 +71,13 @@ end
 
 function rrule(::typeof(cholesky), X::AbstractMatrix{<:Real})
     F = cholesky(X)
-    function cholesky_pullback(Ȳ)
+    function cholesky_pullback(Ȳ::Composite{<:Cholesky})
         # Need to be `Matrix`s for calling `blas` functions in `chol_blocked_rev`.
+        # TODO: no `unthunk`ing - https://github.com/JuliaDiff/ChainRulesCore.jl/issues/100
         ∂X = if F.uplo === 'U'
-            @thunk(chol_blocked_rev(Matrix(Ȳ.U), Matrix(F.U), 25, true))
+            @thunk(chol_blocked_rev(Matrix(unthunk(Ȳ.U)), Matrix(F.U), 25, true))
```

Comment:
Why is `Matrix` required?

Reply:
see comment in the code?

Reply:
blas functions further now complain if not given a StridedMatrix (https://github.com/JuliaDiff/ChainRules.jl/pull/164#discussion_r370601776)

Reply:
Those `Matrix` calls should be inside `chol_blocked_rev` then?

Reply:
done